pkg/services/auth.go

```diff
@@ -6,9 +6,53 @@ import (
 	"go.uber.org/zap"
 
 	"github.com/cmsgov/easi-app/pkg/appcontext"
+	"github.com/cmsgov/easi-app/pkg/graph/model"
 	"github.com/cmsgov/easi-app/pkg/models"
 )
 
+// HasRole authenticates a user as having a given role
+func HasRole(ctx context.Context, role model.Role) (bool, error) {
+	logger := appcontext.ZLogger(ctx)
+	principal := appcontext.Principal(ctx)
+	switch role {
+	case model.RoleEasiUser:
+		if !principal.AllowEASi() {
+			logger.Info("does not have EASi job code")
+			return false, nil
+		}
+		logger.With(zap.Bool("Authorized", true)).
```

Comment:
nit: I see you using the `.With(...).Info(...)` construct. That construct is more for when you're going to be passing around the resultant logger instance (perhaps on the `context`) so that all future log lines are decorated with the `With(...)` data. Since you're not passing actually passing the logger around anywhere, you can just use the normal construct:
`logger.Info("user authorized as EASi user", zap.Bool("Authorized", true))`